### DISCUSSION TITLE
API Error: HTTP Error

### DISCUSSION THREAD

**Discussion by adm2k** [netalertx/NetAlertX#792] [Q&A]
Latest version of NetalertX been running fine until about an hour ago. Getting this:
API Error: HTTP Error
URL: http://192.168.86.24:20211/php/server/devices.php?action=getDevicesTotals

**Comment by jokob-sk** [comment]
Hi @adm2k ,

I just checked and accessing the mentioned URL seems to work on my end:

![image](https://github.com/user-attachments/assets/5b521b0f-e3dc-441f-8365-f4110a7b3174)


are there any errors in the browser console that could help identifying the issue further?

**Comment by adm2k** [comment]
@jokob-sk 

I can't seem to find any errors in the browser (Chrome) console. I restored NetalertX from a previous Proxmox backup to a previous version and the site came up and ran for a few minutes and then started  with the same error. The main NetalertX page, after a few minutes fails to load, and gives an HTTP 500 error.

Thanks, al

**Reply by adm2k** [reply]
@jokob-sk 

Log file is showing the following, it just keeps repeating...

10:44:44 Permissions check (All should be True)
10:44:44 ------------------------------------------------
10:44:44   /config/app.conf |  READ  | True
10:44:44   /config/app.conf |  WRITE | True
10:44:44   /db/app.db       |  READ  | True
10:44:44   /db/app.db       |  WRITE | True
10:44:44 ------------------------------------------------
10:44:44 [Setup] Attempting to fix permissions.
10:44:44 [Setup] Attempting to fix permissions.
10:44:44 [Database] Opening DB
10:44:44 [upgradeDB] Re-creating Settings table
10:44:44 [upgradeDB] Re-creating Pholus_Scan table
10:44:44 [upgradeDB] Re-creating Parameters table
10:44:44 [Config] reading config file
10:44:45 [MAIN] Setting up ...
10:44:45 [conf.tz] Setting up ...
10:44:45 

10:44:45 The container restarted (started). If this is unexpected check https://bit.ly/NetAlertX_debug for troubleshooting tips.
10:44:45

**Comment by adm2k** [comment]
and more from the log...
netalertx-1  |             [INSTALL] 🚀 Starting app (:20211)
netalertx-1  | 
netalertx-1  |         
netalertx-1  | s6-rc: info: service legacy-services successfully started
netalertx-1  | Traceback (most recent call last):
netalertx-1  |   File "<frozen runpy>", line 198, in _run_module_as_main
netalertx-1  |   File "<frozen runpy>", line 88, in _run_code
netalertx-1  |   File "/app/server/__main__.py", line 209, in <module>
netalertx-1  |     sys.exit(main())       
netalertx-1  |              ^^^^^^
netalertx-1  |   File "/app/server/__main__.py", line 76, in main
netalertx-1  |     db.upgradeDB()
netalertx-1  |   File "/app/server/database.py", line 128, in upgradeDB
netalertx-1  | 11:38:43 [MAIN] Setting up ...
netalertx-1  | 11:38:43 [conf.tz] Setting up ...
netalertx-1  | 11:38:43 
netalertx-1  | 
netalertx-1  | 11:38:43 The backend restarted (started). If this is unexpected check https://bit.ly/NetAlertX_debug for troubleshooting tips.
netalertx-1  | 11:38:43

**Reply by jokob-sk** [reply]
Hummm, could you please double check the config and DB folders are accessible and have the right permissions? https://github.com/jokob-sk/NetAlertX/blob/main/docs/FILE_PERMISSIONS.md

It would be great if you could get a more detailed error log if available (e.g. via Portainer logs or by running the container this way, so it doesn't restart and the error is logged on crash in the console:

```
docker run --rm --network=host \
  -v local/path/netalertx/config:/app/config \
  -v local/path/netalertx/db:/app/db \
  -e TZ=Europe/Berlin \
  -e PORT=20211 \
  jokobsk/netalertx:latest
```

**Reply by adm2k** [reply]
@jokob-sk 

I did as you suggested see the attached log file

[log.txt](https://github.com/user-attachments/files/17005328/log.txt)

By the way NetalertX has been running fine for months. This just started after latest upgrade. I restored from a Proxmox backup prior to the upgrade and issue still happens.

Thanks for the help!

**Reply by adm2k** [reply]
@jokob-sk 

I also checked permissions and they seemed correct.

**Reply by jokob-sk** [reply]
Hi @adm2k ,

Thanks for the log. Unfortunately the error doesn't sound like something we can fix:

```
06:05:07 Permissions check (All should be True)
06:05:07 ------------------------------------------------
06:05:07   /config/app.conf |  READ  | True
06:05:07   /config/app.conf |  WRITE | True
06:05:07   /db/app.db       |  READ  | True
06:05:07   /db/app.db       |  WRITE | True
06:05:07 ------------------------------------------------
06:05:07 [Setup] Attempting to fix permissions.
06:05:07 [Setup] Attempting to fix permissions.
06:05:07 [Database] Opening DB
06:05:07 [upgradeDB] Adding dev_Network_Node_MAC_ADDR to the Devices table

            [INSTALL] 🚀 Starting app (:20211)

        
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/app/server/__main__.py", line 209, in <module>
    sys.exit(main())       
             ^^^^^^
  File "/app/server/__main__.py", line 76, in main
    db.upgradeDB()
  File "/app/server/database.py", line 128, in upgradeDB
    self.sql.execute("""
sqlite3.OperationalError: no such table: Devices
```

The `Devices` DB table seems to be missing. The DB might have gotten corrupted during a backup run or the like. Do you have any scripts running for backups? I don't use proxmox so not sure how backups are handled and if R&W operations might intervene with the application runtime.  

In order to restore the app I would suggest to check if you  have a backup of the `devices.csv` file in the `/config` folder. Check this guide for details: https://github.com/jokob-sk/NetAlertX/blob/main/docs/BACKUPS.md 

Please read this carefully to recover most of your setup. You only need the `app.conf` and the aforementioned `devices.csv` file to recover 99% of your setup.

**Reply by adm2k** [reply]
@jokob-sk 

Well after several attempts I managed to recover. I took a copy of the DB and Devices CSV files and copied them off to a safe place. I then completely removed NetalertX and reinstalled and then copied the Db and CSV files back and restored them through the app. All seems to be working now.

Thanks for your feedback, I do appreciate it!

**Reply by jokob-sk** [reply]
Phew, super glad to hear that!
